packages/x-charts/src/PieChart/PieArc.tsx

```diff
@@ -55,42 +58,117 @@ const useUtilityClasses = (ownerState: PieArcOwnerState) => {
   return composeClasses(slots, getPieArcUtilityClass, classes);
 };
 
-const PieArcRoot = styled(animated.path, {
+type PieArcAnimatedProps = Pick<
+  PieArcProps,
+  | 'startAngle'
+  | 'endAngle'
+  | 'cornerRadius'
+  | 'paddingAngle'
+  | 'innerRadius'
+  | 'outerRadius'
+  | 'skipAnimation'
+> & { ref?: React.Ref<SVGPathElement> };
+
+type PieArcInterpolatedProps = Pick<PieArcAnimatedProps, 'startAngle' | 'endAngle'>;
+
+function pieArcPropsInterpolator(from: PieArcInterpolatedProps, to: PieArcInterpolatedProps) {
+  const interpolateStartAngle = interpolateNumber(from.startAngle, to.startAngle);
+  const interpolateEndAngle = interpolateNumber(from.endAngle, to.endAngle);
+
+  return (t: number) => {
+    return {
+      startAngle: interpolateStartAngle(t),
+      endAngle: interpolateEndAngle(t),
+    };
+  };
+}
+
+/** Animates a pie slice using a `path` element.
+ * The props object also accepts a `ref` which will be merged with the ref returned from this hook. This means you can
+ * pass the ref returned by this hook to the `path` element and the `ref` provided as argument will also be called. */
+export function useAnimatePieArc(props: PieArcAnimatedProps): {
```

Comment:
Same as the line, move to `UseAnimatePieParams/ReturnValue` and it would probably be best if these were in their own files or under `hooks/` if the goal is to make them public eventually

Reply:
> if the goal is to make them public eventually

They are already public as part of this PR, so I'll move them there.